### DISCUSSION TITLE
Unable to resolve dependency [Parameter #0 [ <required> $data ]] in class App\Livewire\Component\Connect\ConnectWithData

### DISCUSSION THREAD

**Discussion by maheshharsha** [livewire/livewire#9048] [1. Help]
public $accessToken, $phoneNumberId, $wabaId, $data;
    protected $listeners = ['connectWithFacebook' => 'handleconnectWithFacebook'];

    /**
     * Connect with facebook
     *
     * @return void
     */
    public function  handleconnectWithFacebook($data)
    {
        dd($data);
       
    }
    
    this is my component file 
    <script>
    try{
    Livewire.dispatch('connectWithFacebook', { key: 'value' });
}
</script>
and this is my blade file code
I am facing this issue
Unable to resolve dependency [Parameter #0 [ <required> $data ]] in class App\Livewire\Component\Connect\ConnectWithData

**Comment by joshhanley** [comment]
@maheshharsha your method parameter needs to match the names in your object in the dispatch.So you've called it `key` so your method signature should be `handleconnectWithFacebook($key)`.

**Comment by lrljoe** [comment]
What Josh said;

but while you're at it, I'd suggest you look at the newer approaches for event listeners (the On attribute) in the docs
